src/workerd/api/r2-bucket.h

```diff
@@ -374,12 +374,23 @@ class R2Bucket: public jsg::Object {
     }
   }
 
+  enum class EtagType {
+    Strong,
+    Weak,
+    Wildcard,
+  };
+
+  struct Etag {
+    EtagType type;
+    kj::Maybe<kj::String> value;
+  };
```

Comment:
This works but something like the following would probably be more idiomatic for our codebase:

```cpp
struct WildcardEtag {};
struct WeakEtag {
  kj::String value;
};
struct StrongEtag {
  kj::String value;
};
using Etag = kj::OneOf<WildcardEtag, WeakEtag, StrongEtag>;
```

Reply:
Thank you! That's much closer to how enums work in Rust! That's a great suggestion, will put that in place. It'll let me avoid the maybe on the value, simplifying my code :smile: